### ISSUE TITLE
Have nx format:write sort paths and projects in angular.json, nx.json, and tsconfig.json

### ISSUE THREAD

**Issue by spyderz42** [nrwl/nx#2607]
## Expected Behavior

Currently when adding new components through different branches, conflicts arise as they converge into master/development. If components are sorted by name, then those conflicts will be minimized as it is unlikely that the new components will all be added at the end of the file like they currently are.

## Current Behavior

Not sorted, just added to the end of the list

**Comment by jsaguet** [comment]
I agree that having projects ordered alphabetically in those files would help a lot to avoid conflicts.

I am not sure though if format:write is the appropriate command to do this.
Maybe the generate schematics could implement this behavior to have the projects ordered as soon as they are created?

**Comment by spyderz42** [comment]
Agreed! I would assume that the format:write would use whatever is implemented elsewhere, since I would need it for my current projects to update them.

**Comment by jsaguet** [comment]
I think format command basically runs prettier.

IMHO, alphabetical order should be the default for generate commands but users should have the choice to opt out if they need to order them in a specific way.

It would be cool to have maintainers' opinion on this feature.

I'd be interested to submit a PR for this

**Comment by patdx** [comment]
I tried this for a nx project, I wrote some scripts that will sort angular.json and nx.json and plugged it into husky and lint-staged to format on git pre-commit hook. It works pretty well to avoid merge conflicts and keep the files readable in my opinion.

Unfortunately there's some format thrashing between nx generate commands and the format scripts, might be related to my prettier config but I'm not sure, Also, one of the developers I work with never seem to have git pre-commit hooks working (despite many reminders, haha).

> I am not sure though if format:write is the appropriate command to do this.
> Maybe the generate schematics could implement this behavior to have the projects ordered as soon as they are created?

Yeah, at least for me, some kind of built in solution, like inserting the lines in the correct place the first time, or a hook after the generate command to format the whole file, would be a really nice upgrade for my project.
